src/map/script_map.cpp

```diff
@@ -1056,6 +1056,71 @@ static int CclGetIsGameHoster(lua_State *l)
 	return 1;
 }
 
+/**
+** <b>Description</b>
+**
+** Set basic map caracteristics.
+**
+**  @param l  Lua state.
+**
+** Example:
+**
+** <div class="example"><code><strong>PresentMap</strong>("Map description", PlayerCount, Width, Height, uid_number [, "highgrounds-enabled"])</code></div>
+*/
+static int CclPresentMap(lua_State *l)
+{
+	LuaCheckArgs_min(l, 5);
+
+	Map.Info.Description = LuaToString(l, 1);
+	// Number of players in LuaToNumber(l, 2); // Not used yet.
+	Map.Info.MapWidth = LuaToNumber(l, 3);
+	Map.Info.MapHeight = LuaToNumber(l, 4);
+	Map.Info.MapUID = LuaToNumber(l, 5);
+	
+	if(LuaGetArgsNum(l) >= 6) {
+		const std::string_view value = LuaToString(l, 6);
+		if (value == "highgrounds-enabled") {
+			Map.Info.EnableHighgrounds();
+		} else {
+			LuaError(l, "Unknown value %s\n", value.data());
+		}
+	}
+
+	return 0;
+}
+
+static int CclMapEnableHighgrounds(lua_State *l)
+{
+	Map.Info.EnableHighgrounds(LuaGetArgsNum(l) >= 1 ? LuaToBoolean(l, 1) : true);
+	
+	return 0;
+}
+
+static int CclIsHighgroundsEnabled(lua_State *l)
+{
+	lua_pushboolean(l, Map.Info.IsHighgroundsEnabled());
+	return 1;
+}
```

Comment:
Might probably go in map.pkg (tolua++) instead.

Reply:
What's the difference in principle? 

I just did it by analogy, depending on how existing functions with similar functionality were translated into lua. If, for example, what concerns gui was translated through 'tolua++', then almost everything I've encountered concerning engine settings or parameters of the current game was done through `lua_register()` . 

Changing the method is not a problem. I just want to understand for the future on what principle to choose the translation method.

Reply:
Not sure neither ;-)

to_lua has been added after.

BTW, the flag seems unused in the Engine :-/ ...
Is there incoming PRs next?
How does it work?

Reply:
It used here: https://github.com/Wargus/wargus/pull/446

Reply:
According to this flag we load base or run extended tileset generator to make tiles for highgrounds/ramps/cliffs etc.

Reply:
Currently
```
      Map.Info.Postamble    = ""
      MapEnableHighgrounds(highgroundsCheckBox:isMarked())
```
With Tolua, it would be more consistent:
```
      Map.Info.Postamble    = ""
      Map.Info.HighgroundsEnabled = highgroundsCheckBox:isMarked()
```
but it seems that Highgrounds is more related to tileset than map.
wonder if it is then the right flag/place.
Shouldn't we choose between summer/summer_highgrounds/swamp/swamp_highgrounds...?

Reply:
>Map.Info.HighgroundsEnabled = highgroundsCheckBox:isMarked()

This is a private member and I don't want to open it in case if we will need to do something else with enabling it in the future. At least the network synchronization may be required. 

I could use Tolua to open `MapEnableHighgrounds()`/`IsHighgroundsEnabled()`, but definitely don't want to make `Map.Info.HighgroundsEnabled` variable as public.

Reply:
> Shouldn't we choose between summer/summer_highgrounds/swamp/swamp_highgrounds...?

Don't breed entities. )

The player simply selects the basic tileset as before, and wargus chooses whether to load the extended version or not. Moreover, if wargus "sees" that the map contains highgrounds, it will simply hide the possibility to select (in dropDownList) the base tileset, for which there is no possibility to generate the extended version (I am talking about swamp specifically - there are certain difficulties with generation).

For the editor, there is a checkbox to enable highgrounds or not.

So the `Map.Info.HighgroundsEnabled` is map's property, on the basis of which the choice of which tileset to load is made

Reply:
I've posted a visual explanation: https://discord.com/channels/780082494447288340/987001693474541570/1179091461820666008

Reply:
Also the map presentation is loaded before tileset. Tileset loads with game start, but map presentation is uploaded to the stage of choosing the game parameters by the player (in the create a new game menu).